framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -1105,6 +1135,27 @@ public ClassOrInterfaceAnnos(@Nullable TypeDeclaration<?> javaParserNode) {
       classDeclaration = javaParserNode;
     }
 
+    /**
+     * Returns a deep copy of this.
+     *
+     * @return a deep copy of this
+     */
+    @Override
+    public ClassOrInterfaceAnnos deepCopy() {
+      try {
+        ClassOrInterfaceAnnos result = (ClassOrInterfaceAnnos) super.clone();
```

Comment:
I don't think I would use clone at all since it's specification is so fuzzy.

Reply:
The goal of calling `clone()` is to create a new object with exactly the same class as the receiver, and with fields initialized.  It is safe for that purpose.  An advantage of `clone()` is that if a declaration changes or a new superclass is added, then clone() still works and still produces an object of that type with fields defaulted (here, `classDeclaration` is relevant).  Furthermore, every clone() method definition starts with super.clone() and so doing the same thing in other copy methods feels right to me.

If you don't like its use even in that situation, or think it is too error-prone, I can change it.  In that case, I would create a private constructor that takes enough arguments for all the fields, ensuring that each one is set explicitly (rather than having `deepCopy()` do it.

Reply:
I also I'm just imagining the nightmare of debugging a problem where the copy returned by `deepCopy` isn't deep enough and so mutations are happening on the original. Or the fact that a new object is created without any constructor being invoked.  Though I don't think either of those things are going to happen in this case.  And we are only talking about a few classes that aren't used through-out the framework.

I think in the end, what was most concerning about marking `AnnotatedTypeMirror`s `Clonable`, and since you changed that, I think this limited use of clone is ok.